src/Bundle/DependencyInjection/Compiler/RegisterFQCNControllersPass.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ResourceBundle\DependencyInjection\Compiler;
+
+use Sylius\Bundle\ResourceBundle\Controller\ResourceController;
+use Sylius\Component\Resource\Metadata\Metadata;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/** @internal */
+final class RegisterFQCNControllersPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        try {
+            $resources = $container->getParameter('sylius.resources');
+        } catch (InvalidArgumentException $exception) {
+            return;
+        }
+
+        foreach ($resources as $alias => $configuration) {
+            $metadata = Metadata::fromAliasAndConfiguration($alias, $configuration);
+
+            $this->validateSyliusResource($metadata->getClass('model'));
+            $controllerFQCN = $metadata->getClass('controller');
+
+            if ($controllerFQCN !== ResourceController::class) {
+                $definition = $container->getDefinition($metadata->getServiceId('controller'));
+
+                // TODO: Change to alias definition after bumping to > Symfony 5.2
```

Comment:
Why can't we use aliases in Symfony 4.4?

Reply:
Because we need here public aliases which will be provided in Symfony [5.2](https://github.com/symfony/http-kernel/commit/06d999d7a5ade9843bfb613340493ebd752c69d0)

Reply:
It's available in Symfony 5.1 as well.

Reply:
Could we change the comment? Or maybe put some if to find out if Symfony 5.1 is installed and then using an alias instead of copied definition?